automation/Extension/firefox/feature.js/callstack-instrument.js

```diff
@@ -0,0 +1,19 @@
+/*
+    This script catches the stacktraces of JavaScript that ran on a crawled site and made a WebRequest
+    and saves them to the "callstacks" table
+*/
+export class CallstackInstrument {
+    constructor(dataReceiver) {
+        this.dataReceiver = dataReceiver;
+    }
+    run(crawlID) {
+        browser.stackDump.onStackAvailable.addListener((requestId, stack) => {
```

Comment:
If we use `crawl_id`, `request_id` and `call_stack` as the parameter names, then we can use shorthand notation in the object:
```
this.dataReceiver.saveRecord("callstacks", { crawl_id, request_id, call_stack });
```